pkg/daemon/ceph/osd/volume.go

```diff
@@ -282,6 +282,16 @@ func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *Device
 				deviceArg,
 			}...)
 
+			if a.replaceOSD != nil {
+				replaceOSDID := a.GetRestoreOSDId(device.DeviceInfo.RealPath)
+				if replaceOSDID != -1 {
+					immediateExecuteArgs = append(immediateExecuteArgs, []string{
+						"--osd-id",
+						fmt.Sprintf("%d", replaceOSDID),
```

Comment:
Why do we need to look up the ID on line 286? Can't we just set the ID directly? Or is this to allow other OSDs in the same OSD prepare job to be evaluated? Seems like we want to skip all other OSDs anyway.
```suggestion
						fmt.Sprintf("%d", a.replaceOSD.ID),
```

Reply:
We are preparing OSDs in a loop. So validating the ID once more to ensure that `--osd-id` is used with the correct `device` every time.  
You are right about the others OSDs. They will be skipped. But there can a scenario that a new disk was added and it got discovered.